packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -123,39 +91,33 @@ class Firestore {
       {Duration timeout = const Duration(seconds: 5)}) async {
     assert(timeout.inMilliseconds > 0,
         'Transaction timeout must be more than 0 milliseconds');
-    final int transactionId = _transactionHandlerId++;
-    _transactionHandlers[transactionId] = transactionHandler;
-    final Map<String, dynamic> result = await channel
-        .invokeMapMethod<String, dynamic>(
-            'Firestore#runTransaction', <String, dynamic>{
-      'app': app.name,
-      'transactionId': transactionId,
-      'transactionTimeout': timeout.inMilliseconds
-    });
+
+    // Wrap the transactionHandler into something that can be passed to the Platform implementation
+    final PlatformTransactionHandler handler = (int transactionId) async {
+      Transaction transaction = Transaction(transactionId, this);
+      final dynamic result = await transactionHandler(transaction);
+      await transaction._finish();
+      return result;
+    };
+
+    // Move to a runTransaction method in the method channel!
```

Comment:
Is this comment intended to be a TODO(ditman)? If so I would format it that way and remove the !

Also maybe use backticks around method names like `runTransaction` but that is really nitpicking

Reply:
Nope! This is a lie now, forgot to remove it. Cleaning.